spec/scnnr/client_spec.rb

```diff
@@ -39,6 +39,54 @@
     end
   end
 
+  shared_examples 'posting an image' do
+    context 'when the timeout is larger than the API supports' do
+      let(:api_max_timeout) { Scnnr::PollingManager::MAX_TIMEOUT }
+      let(:timeout) { api_max_timeout * 3 + 1 }
+      let(:uri) { an_object_having_attributes(query: a_string_matching(/timeout=#{api_max_timeout}/)) }
+
+      let(:queued_recognition) { Scnnr::Recognition.new('id' => 'queued_id', 'state' => 'queued') }
+      let(:finished_recognition) { Scnnr::Recognition.new('id' => 'finished_id', 'state' => 'finished') }
+
+      context 'when the first request finishes' do
+        before do
+          expect(mock_response).to receive(:build_recognition) { finished_recognition }
+        end
+
+        it 'immediately returns the recognition' do
+          expect(client).not_to receive(:fetch)
+          expect(subject).to eq finished_recognition
+        end
+      end
+
+      context 'when the second attempt finishes' do
+        before do
+          expect(mock_response).to receive(:build_recognition) { queued_recognition }
```

Comment:
Thanks for adding tests! It is excellent, but I have something to say.
I've used `expect` and `allow` as following.

```
expect -> it works as a mock and verifies method calling, so it should be used in `it` block.
allow -> it works as a stub and verifies nothing, so it should be used in `before` block.
```

so I think all `expect` checks are used in `it` block. 
what do you think?

Reply:
Good point. I've become used to using `expect` by default because it's occasionally useful to know when you incorrectly assume something will only be called once. On the other hand, changing from `expect` to `allow` showed me some places where I had valuable assertions in my `before` blocks, so I've changed it here and moved those assertions to `it` blocks.